### ISSUE TITLE
MIT Licensing

### ISSUE THREAD

**Issue by sepp87** [opensourceBIM/BIMsurfer#92]
So BIMsurfer has the MIT License, but it loads its models from BIMserver. BIMserver has an AGPL license. Doesn't BIMsurfer need to have the same license since it links to BIMserver?

https://softwareengineering.stackexchange.com/questions/107883/agpl-what-you-can-do-and-what-you-cant

**Comment by hlg** [comment]
No, BIMsurfer does not use any BIMserver code (note the difference between loading data and loading code). Like all the clients, it can have any license, including closed-source licenses. The notable exception are Java clients using the BimServerClientLib module - they would need to be under GPL license.

**Comment by jovancusic** [comment]
I'm looping myself in since we were exploring using BIMServer as well and decided against it:
The problem is basically this: if you link against an AGPL component, your own component also needs to be AGPL. “Linking” isn’t only static/dynamic linking. With AGPL it also includes non-trivial network calls — e.g. websockets, RPC-style REST, custom APIs, anything where both pieces form one “combined service”.
For MIT-licensed code that’s fine, because MIT lets you relicense under AGPL.
 But for commercial code or copyleft licenses that are not AGPL-compatible, it’s a big problem — you can’t relicense them, so the whole setup becomes legally incompatible.
Google recently wrote about this in a post that went viral: https://opensource.google/documentation/reference/using/agpl-policy

**Comment by hlg** [comment]
No, the definition of linking is the same for GPL and AGPL. It is the definition of distribution that makes the difference. As long as you don't distribute the code or program, you don't have any obligations from the GPL or AGPL. With the AGPL, distribution over the network is relevant - which is not the case for the GPL. BIMsurfer is not linking to BIMserver and thus there is no reason for its license to be compatible with the AGPL, even if it connects over the network. I believe that much of the fear of using code licensed under GPL or AGPL stems from misconceptions like these. 

Btw. "RPC-style REST" sounds like a GPT fever dream to me, and some other parts of the post as well.
